clang/test/Driver/sycl-ftarget-register-alloc-mode.cpp

```diff
@@ -0,0 +1,77 @@
+// Test SYCL -ftarget-register-alloc-mode
+
+// RUN: %clang -### -fsycl \
+// RUN:    -fsycl-targets=spir64_gen -ftarget-register-alloc-mode=pvc:auto %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=AUTO_AOT %s
+
+// RUN: %clang -### -fsycl \
+// RUN:    -fsycl-targets=spir64_gen -ftarget-register-alloc-mode=pvc:large %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=LARGE_AOT %s
+
+// RUN: %clang -### -fsycl \
+// RUN:    -fsycl-targets=spir64_gen -ftarget-register-alloc-mode=pvc:small %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=SMALL_AOT %s
+
+// RUN: %clang -### -fsycl \
+// RUN:    -fsycl-targets=spir64_gen -ftarget-register-alloc-mode=pvc:default %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=DEFAULT_AOT %s
+
+// RUN: %clang -### -fsycl \
+// RUN:    -fsycl-targets=spir64_gen -ftarget-register-alloc-mode=pvc:small,pvc:large %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=MULTIPLE_ARGS_AOT %s
+
+// RUN: %clang -### -fsycl \
+// RUN:   -ftarget-register-alloc-mode=pvc:auto %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=AUTO_JIT %s
+
+// RUN: %clang -### -fsycl \
+// RUN:   -ftarget-register-alloc-mode=pvc:large %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=LARGE_JIT %s
+
+// RUN: %clang -### -fsycl \
+// RUN:   -ftarget-register-alloc-mode=pvc:small %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=SMALL_JIT %s
+
+// RUN: %clang -### -fsycl \
+// RUN:   -ftarget-register-alloc-mode=pvc:default %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=DEFAULT_JIT %s
+
+// RUN: %clang -### -fsycl \
+// RUN:   -ftarget-register-alloc-mode=pvc:small,pvc:large %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=MULTIPLE_ARGS_JIT %s
```

Comment:
Could you add some negative tests to verify bad argument passing according to your conditions on L991 in sycl.cpp?

Reply:
Done in latest commit, thanks!